vfs/mem_fs.go

```diff
@@ -657,22 +652,24 @@ func (f *memNode) resetToSyncedState() {
 	}
 }
 
-// memFile is a reader or writer of a node's data, and implements File.
+// memFile is a reader or writer of a node's data. It implements os.FileInfo and
+// File interfaces.
 type memFile struct {
+	name        string
 	n           *memNode
 	fs          *MemFS // nil for a standalone memFile
 	rpos        int
 	wpos        int
 	read, write bool
 }
 
+var _ os.FileInfo = (*memFile)(nil)
 var _ File = (*memFile)(nil)
 
 func (f *memFile) Close() error {
 	if n := f.n.refs.Add(-1); n < 0 {
 		panic(fmt.Sprintf("pebble: close of unopened file: %d", n))
 	}
-	f.n = nil
```

Comment:
Needed to remove this line because `memFile.Stat()` must return `FileInfo` that can outlive the `memFile` after its `Close()`. Without this line `TestCheckpoint` panics.

Reply:
I have a PR out for this.